### PR TITLE
feat: Show similar events based on event topic

### LINKED ISSUE #123: Compare Event Topics and show a list of similar events in event details
For now, we can show the user a list of similar events in the event details section by showing events based on the event topic
As in eventbrite
screenshot_20180613-015207
Working on it :+1:

### PR DESCRIPTION
**Pull request by dreadpool2**
Fixes #123 
videotogif_2018 06 25_02 31 29

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/similar/EventTopicId.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package org.fossasia.openevent.general.similar
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/similar/EventTopicId.kt]: What similar?

File: app/src/main/java/org/fossasia/openevent/general/similar/EventTopicsDao.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+*/
+    @Query("SELECT * from Event WHERE eventTopic = :topicId")
+    fun getAllSimilarEvents(topicId: Long): Flowable<List<Event>>
+
+    /*
+    @Query("SELECT * from EventTopic WHERE id = :eventId")
+    fun getEventTopicOfEvent(eventId: Long): Flowable<EventTopic>
+*/
+    @Query("UPDATE Event SET favorite = :favorite WHERE id = :eventId")
+    fun setFavorite(eventId: Long, favorite: Boolean)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/similar/EventTopicsDao.kt]: Ummm This is EventDao

File: app/src/main/java/org/fossasia/openevent/general/similar/SimilarEventsViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+class SimilarEventsViewModel(private val eventTopicService: EventTopicService) : ViewModel() {
+
+    private val compositeDisposable = CompositeDisposable()
+
+    val progress = MutableLiveData<Boolean>()
+    val similarEvents = MutableLiveData<List<Event>>()
+    val error = MutableLiveData<String>()
+
+    fun getSimilarEvents(id: Long) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/similar/SimilarEventsViewModel.kt]: A void getter!!!

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/similar/SimilarEventsViewModel.kt]: Will change the function name :smile:

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/similar/SimilarEventsViewModel.kt]: Updated PR :+1:

File: app/src/main/java/org/fossasia/openevent/general/event/EventsRecyclerAdapter.kt [hunk 4]
```diff
@@ -19,15 +21,24 @@ class EventsRecyclerAdapter : RecyclerView.Adapter<EventViewHolder>() {
...
         if (events.isNotEmpty())
             this.events.clear()
         this.events.addAll(eventList)
     }
 
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): EventViewHolder {
-        val view = LayoutInflater.from(parent.context).inflate(R.layout.item_card_events, parent, false)
-        return EventViewHolder(view)
+        val eventView: View
+        if (eventLayout.equals("similarEvents")) {
```
Comment (simarsingh24) [inline app/src/main/java/org/fossasia/openevent/general/event/EventsRecyclerAdapter.kt]: dont use exact matched keep a constant somewhere and reference it from there

File: app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicApi.kt [hunk 5]
```diff
@@ -17,5 +18,5 @@ interface EventTopicApi {
 
     @GET("event-topics/{id}")
     fun getEventTopic(@Path("id") id: Long): Single<EventTopic>
-
+*/
```
Comment (simarsingh24) [inline app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicApi.kt]: why are you keeping commented code?

File: app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicService.kt [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        val eventsFlowable = eventTopicsDao.getAllSimilarEvents(id)
+        return eventsFlowable.switchMap {
+            if (it.isNotEmpty())
+                eventsFlowable
+            else
+                eventApi.getEventsUnderTopicId(id)
+                        .map {
+                            eventTopicsDao.insertSimilarEvents(it)
+                        }
+                        .toFlowable()
```
Comment (simarsingh24) [inline app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicService.kt]: take this to the line above

File: app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicService.kt [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        }
+    }
+/*
+    fun getEventTopicOfEvent(id: Long): Flowable<EventTopic> {
+        return eventApi.getEventTopicOfEvent(id)
+                        .map {
+                            eventTopicsDao.insertEventTopic(it)
+                            it
+                        }
+                        .toFlowable()
```
Comment (simarsingh24) [inline app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicService.kt]: same here

File: app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicsDao.kt [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+import org.fossasia.openevent.general.event.Event
+
+@Dao
+interface EventTopicsDao {
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    fun insertSimilarEvents(events: List<Event>)
+/*
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    fun insertEventTopic(eventTopic: EventTopic)
+*/
```
Comment (simarsingh24) [inline app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicsDao.kt]: commented code?

File: app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicsDao.kt [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+    @Query("DELETE FROM Event")
+    fun deleteAllEvents()
+/*
+    @Query("DELETE FROM EventTopic")
+    fun deleteAllEventTopics()
+*/
```
Comment (simarsingh24) [inline app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicsDao.kt]: ?

File: app/src/main/java/org/fossasia/openevent/general/similarEvents/SimilarEventsFragment.kt [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+        }
+
+        val favouriteFabClickListener = object : FavoriteFabListener {
+            override fun onClick(eventId: Long, isFavourite: Boolean) {
+                similarEventsViewModel.setFavorite(eventId, !isFavourite)
+            }
+        }
+
+        similarEventsRecyclerAdapter.setListener(recyclerViewClickListener)
+
```
Comment (simarsingh24) [inline app/src/main/java/org/fossasia/openevent/general/similarEvents/SimilarEventsFragment.kt]: extra line feed

File: app/src/main/java/org/fossasia/openevent/general/similarEvents/SimilarEventsViewModel.kt [hunk 11]
```diff
@@ -0,0 +1,55 @@
...
+                    Timber.e(it, "Error")
+                    error.value = "Error"
+                }))
+    }
+
+    override fun onCleared() {
+        super.onCleared()
+        compositeDisposable.clear()
+    }
+
```
Comment (simarsingh24) [inline app/src/main/java/org/fossasia/openevent/general/similarEvents/SimilarEventsViewModel.kt]: extra line

File: app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicsDao.kt (hunk 9 above)
Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicsDao.kt]: I thought I might probably require these methods in future, will probably remove them for now :+1:

**Comment by simarsingh24** [comment]
good work btw 👍

File: app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicsDao.kt (hunk 9 above)
Comment (simarsingh24) [reply app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicsDao.kt]: yes please remove this dont keep commented code in final version

File: app/src/main/java/org/fossasia/openevent/general/similarEvents/SimilarEventsViewModel.kt [hunk 12]
```diff
@@ -0,0 +1,55 @@
...
+class SimilarEventsViewModel(private val eventTopicService: EventTopicService) : ViewModel() {
+
+    private val compositeDisposable = CompositeDisposable()
+
+    val progress = MutableLiveData<Boolean>()
+    val similarEvents = MutableLiveData<List<Event>>()
+    val error = MutableLiveData<String>()
+
+    fun loadSimilarEvents(id: Long) {
+        if(id.toString() == "-1"){
```
Comment (nikit19) [inline app/src/main/java/org/fossasia/openevent/general/similarEvents/SimilarEventsViewModel.kt]: why a conversion to string ? You can compare that directly

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/similarEvents/SimilarEventsViewModel.kt]: It's not working if I use id.equals(-1).

Comment (simarsingh24) [reply app/src/main/java/org/fossasia/openevent/general/similarEvents/SimilarEventsViewModel.kt]: why dont you do id == -1

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/similarEvents/SimilarEventsViewModel.kt]: id is of type long actually, so an error is cropping up

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/similarEvents/SimilarEventsViewModel.kt]: Updated PR Please review :+1:  . Didn't change the string conversion

**Comment by dreadpool2** [comment]
@iamareebjamal Please Review :+1:

**Comment by dreadpool2** [comment]
@iamareebjamal Could you please review this as I want to work on #207 after this

File: app/src/main/java/org/fossasia/openevent/general/event/Event.kt [hunk 13]
```diff
@@ -62,5 +65,8 @@ data class Event(
         val isTicketingEnabled: Boolean = false,
         val isTaxEnabled: Boolean = false,
         val isMapShown: Boolean = false,
-        val favorite: Boolean = false
+        val favorite: Boolean = false,
+        @ColumnInfo(index = true)
+        @Relationship("event-topic")
+        var eventTopic: EventTopicId? = null
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/event/Event.kt]: Why not load the entire event topic in one go?

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/event/Event.kt]: This way you can even show the event topic's name

File: app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt [hunk 14]
```diff
@@ -99,6 +103,11 @@ class EventDetailsFragment : Fragment() {
         rootView.startsOn.text = EventUtils.getFormattedDate(startsAt)
         rootView.endsOn.text = EventUtils.getFormattedDate(endsAt)
 
+        //Set event topic id
+        if(event.eventTopic != null){
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt]: Space around symbols

File: app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt [hunk 15]
```diff
@@ -99,6 +103,11 @@ class EventDetailsFragment : Fragment() {
...
+            eventTopicId = (event.eventTopic)?.id
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt]: No need for parenthesis

File: app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt [hunk 16]
```diff
@@ -245,6 +254,17 @@ class EventDetailsFragment : Fragment() {
...
     }
+
+    private fun loadSimilarEventsFragment(){
+        //Initialise SimilarEvents Fragment
+        val similarEventsFragment = SimilarEventsFragment()
+        val bundle = Bundle()
+        bundle.putLong(EVENT_ID, eventId)
+        eventTopicId?.let { bundle.putLong(EVENT_TOPIC_ID, it) }
+        similarEventsFragment.arguments = bundle
+        val transaction = childFragmentManager.beginTransaction()
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt]: No need of storing transaction in a variable

File: app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt [hunk 17]
```diff
@@ -27,6 +27,10 @@ class EventsFragment : Fragment() {
     private val eventsViewModel by viewModel<EventsViewModel>()
     private lateinit var rootView: View
 
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        eventsRecyclerAdapter.setEventLayout(EventUtils.EVENTS)
+    }
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: New line after }

File: app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicIdConverter.kt [hunk 18]
```diff
@@ -0,0 +1,20 @@
+package org.fossasia.openevent.general.similarEvents
+
+import android.arch.persistence.room.TypeConverter
+
+class EventTopicIdConverter {
+
+    @TypeConverter
+    fun fromEventTopicId(eventTopicId: EventTopicId?): Long?{
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicIdConverter.kt]: Space before {

File: app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicService.kt [hunk 19]
```diff
@@ -0,0 +1,32 @@
+package org.fossasia.openevent.general.similarEvents
+
+import io.reactivex.Completable
+import io.reactivex.Flowable
+import org.fossasia.openevent.general.event.Event
+import org.fossasia.openevent.general.event.EventDao
+
+class EventTopicService(private val eventApi: EventTopicApi, private val eventTopicsDao: EventTopicsDao, private val eventDao: EventDao) {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicService.kt]: Property per line

File: app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicService.kt [hunk 20]
```diff
@@ -0,0 +1,32 @@
...
+    fun getSimilarEvents(id: Long): Flowable<List<Event>> {
+        val eventsFlowable = eventTopicsDao.getAllSimilarEvents(id)
+        return eventsFlowable.switchMap {
+            if (it.isNotEmpty())
+                eventsFlowable
+            else
+                eventApi.getEventsUnderTopicId(id)
+                        .toFlowable()
+                        .map {
+                            eventTopicsDao.insertSimilarEvents(it)
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicService.kt]: How is this different from event service's method?

File: app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicService.kt [hunk 21]
```diff
@@ -0,0 +1,32 @@
...
+                        }
+                        .flatMap {
+                            eventsFlowable
+                        }
+        }
+    }
+
+    fun setFavorite(eventId: Long, favourite: Boolean): Completable {
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicService.kt]: Why create unnecessary duplicate methods?

File: app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicsDao.kt [hunk 22]
```diff
@@ -0,0 +1,20 @@
...
+@Dao
+interface EventTopicsDao {
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    fun insertSimilarEvents(events: List<Event>)
+
+    @Query("DELETE FROM Event")
+    fun deleteAllEvents()
+
+    @Query("SELECT * from Event WHERE eventTopic = :topicId")
+    fun getAllSimilarEvents(topicId: Long): Flowable<List<Event>>
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicsDao.kt]: Add this in EventDao, you are selecting from the same table, so should remain in the same DAO

By this logic, we should have FavoriteDao as well

File: app/src/main/java/org/fossasia/openevent/general/similarEvents/SimilarEventsViewModel.kt (hunk 12 above)
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/similarEvents/SimilarEventsViewModel.kt]: Use id == -1

File: app/src/main/java/org/fossasia/openevent/general/event/Event.kt (hunk 13 above)
Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/event/Event.kt]: Great! I will do that :+1:

**Comment by dreadpool2** [comment]
PR Updated :+1:

File: app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicApi.kt [hunk 23]
```diff
@@ -1,12 +1,13 @@
...
 
 import io.reactivex.Single
+import org.fossasia.openevent.general.event.Event
 import retrofit2.http.GET
 import retrofit2.http.Path
 
 interface EventTopicApi {
 
-    @GET("event-topics/{id}/events")
+    @GET("event-topics/{id}/events?include=event-topic&fields[event-topic]=id&page[size]=0")
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicApi.kt]: No need to restrict the event topic field now

File: app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicsDao.kt [hunk 24]
```diff
@@ -0,0 +1,18 @@
...
+
+@Dao
+interface EventTopicsDao {
+    @Insert(onConflict = OnConflictStrategy.REPLACE)
+    fun insertSimilarEvents(events: List<Event>)
+
+    @Query("DELETE FROM Event")
+    fun deleteAllEvents()
+
+}
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicsDao.kt]: No need of this file

File: app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopic.kt [hunk 25]
```diff
@@ -0,0 +1,13 @@
...
+import com.github.jasminb.jsonapi.LongIdHandler
+import com.github.jasminb.jsonapi.annotations.Id
+import com.github.jasminb.jsonapi.annotations.Type
+
+@Type("event-topic")
+data class EventTopic(
+        @Id(LongIdHandler::class)
+        val id: Long? = null,
+        val name: String? = null,
+        val slug: String? = null
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopic.kt]: Add a foreign key to event

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopic.kt]: Actually EventTopic is not being inserted into the database, so I thought that foriegn key is not needed

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopic.kt]: Okay! Got it wrong :+1:

Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopic.kt]: You have to insert it into database as well

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopic.kt]: Updated PR :+1:

File: app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopic.kt [hunk 26]
```diff
@@ -0,0 +1,19 @@
...
+import org.fossasia.openevent.general.event.Event
+
+@Type("event-topic")
+@Entity(foreignKeys = [(ForeignKey(entity = Event::class, parentColumns = ["id"], childColumns = ["id"], onDelete = ForeignKey.CASCADE))])
+data class EventTopic(
+        @Id(LongIdHandler::class)
+        @PrimaryKey
+        val id: Long?,
+        val name: String?,
+        val slug: String?
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopic.kt]: There's no eventId to make foreign key of

**Comment by dreadpool2** [comment]
@iamareebjamal Did all the changes requested :+1:

File: app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicsDao.kt (hunk 24 above)
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicsDao.kt]: No need of this file

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopicsDao.kt]: Won't we require this in future to show users a list of event topics ?

**Comment by dreadpool2** [comment]
Removed EventTopicsDao for while :+1:

File: app/src/main/java/org/fossasia/openevent/general/event/EventTopic.kt [hunk 27]
```diff
@@ -1,19 +0,0 @@
...
-import com.fasterxml.jackson.databind.PropertyNamingStrategy
-import com.fasterxml.jackson.databind.annotation.JsonNaming
-import com.github.jasminb.jsonapi.LongIdHandler
-import com.github.jasminb.jsonapi.annotations.Id
-import com.github.jasminb.jsonapi.annotations.Relationship
-import com.github.jasminb.jsonapi.annotations.Type
-
-@Type("event-topic")
-@JsonNaming(PropertyNamingStrategy.KebabCaseStrategy::class)
-data class EventTopic(
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/event/EventTopic.kt]: Why did you remove this?

File: app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopic.kt [hunk 28]
```diff
@@ -0,0 +1,25 @@
+package org.fossasia.openevent.general.similarEvents
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopic.kt]: Package name doesn't have camel case

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/similarEvents/EventTopic.kt]: Changed it to `similar_events`

File: app/src/main/java/org/fossasia/openevent/general/event/EventUtils.kt [hunk 29]
```diff
@@ -10,6 +10,9 @@ import timber.log.Timber
 
 object EventUtils {
 
+    val SIMILAR_EVENTS: String = "similarEvents"
+    val EVENTS: String = "events"
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/event/EventUtils.kt]: Where are these needed?

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/event/EventUtils.kt]: These are needed in EventsFragment and SimilarEventsFragment respectively. I have used the same adapter and ViewHolder for both the recycler views.

File: app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt [hunk 30]
```diff
@@ -27,6 +27,11 @@ class EventsFragment : Fragment() {
     private val eventsViewModel by viewModel<EventsViewModel>()
     private lateinit var rootView: View
 
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        eventsRecyclerAdapter.setEventLayout(EventUtils.EVENTS)
```
Comment (dreadpool2) [inline app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: @iamareebjamal Here is one case

File: app/src/main/java/org/fossasia/openevent/general/similar_events/SimilarEventsFragment.kt [hunk 31]
```diff
@@ -0,0 +1,85 @@
...
+    private val similarEventsRecyclerAdapter: EventsRecyclerAdapter = EventsRecyclerAdapter()
+    private val similarEventsViewModel by viewModel<SimilarEventsViewModel>()
+    private lateinit var rootView: View
+    private var EVENT_TOPIC_ID: String = "EVENT_TOPIC_ID"
+    private var eventTopicId: Long = -1
+    private lateinit var linearLayoutManager: LinearLayoutManager
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        similarEventsRecyclerAdapter.setEventLayout(EventUtils.SIMILAR_EVENTS)
```
Comment (dreadpool2) [inline app/src/main/java/org/fossasia/openevent/general/similar_events/SimilarEventsFragment.kt]: Here is another case

File: app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt (hunk 30 above)
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: Use them locally then, no need for these in EventUtils

File: app/src/main/java/org/fossasia/openevent/general/similar_events/SimilarEventsFragment.kt (hunk 31 above)
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/similar_events/SimilarEventsFragment.kt]: Underscore is also not allowed, move the package in event package and rename to topic

File: app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt (hunk 30 above)
Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: Okay :+1:

Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: Did the changes :+1:

File: app/src/main/java/org/fossasia/openevent/general/similar_events/SimilarEventsFragment.kt (hunk 31 above)
Comment (dreadpool2) [reply app/src/main/java/org/fossasia/openevent/general/similar_events/SimilarEventsFragment.kt]: Changed package name :+1:

File: app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt [hunk 32]
```diff
@@ -20,14 +20,17 @@ import org.fossasia.openevent.general.utils.nullToEmpty
 import org.koin.android.architecture.ext.viewModel
 import timber.log.Timber
 import android.os.Build
+import org.fossasia.openevent.general.event.topic.SimilarEventsFragment
 import kotlinx.android.synthetic.main.fragment_event.view.*
 import android.support.v4.content.ContextCompat
 
 class EventDetailsFragment : Fragment() {
     val EVENT_ID = "EVENT_ID"
+    val EVENT_TOPIC_ID = "EVENT_TOPIC_ID"
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt]: private const val and above as well

File: app/src/main/java/org/fossasia/openevent/general/event/EventsRecyclerAdapter.kt [hunk 33]
```diff
@@ -2,14 +2,18 @@ package org.fossasia.openevent.general.event
...
 import org.fossasia.openevent.general.R
+import org.fossasia.openevent.general.event.topic.SimilarEventsFragment
 import java.util.*
 
 class EventsRecyclerAdapter : RecyclerView.Adapter<EventViewHolder>() {
     private val events = ArrayList<Event>()
     private var clickListener: RecyclerViewClickListener? = null
     private var favoriteFabListener: FavoriteFabListener? = null
+    private var eventLayout: String? = null
+    val SIMILAR_EVENTS: String = "similarEvents"
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/event/EventsRecyclerAdapter.kt]: private const val

File: app/src/main/java/org/fossasia/openevent/general/event/topic/SimilarEventsFragment.kt [hunk 34]
```diff
@@ -0,0 +1,86 @@
...
+import org.fossasia.openevent.general.utils.Utils
+import org.koin.android.architecture.ext.viewModel
+import timber.log.Timber
+
+class SimilarEventsFragment : Fragment() {
+    private val similarEventsRecyclerAdapter: EventsRecyclerAdapter = EventsRecyclerAdapter()
+    private val similarEventsViewModel by viewModel<SimilarEventsViewModel>()
+    private lateinit var rootView: View
+    private var EVENT_TOPIC_ID: String = "EVENT_TOPIC_ID"
+    val SIMILAR_EVENTS: String = "similarEvents"
```
Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/general/event/topic/SimilarEventsFragment.kt]: Use once

**Comment by iamareebjamal** [comment]
@dreadpool2 Squash your commits

**Comment by dreadpool2** [comment]
@iamareebjamal Thanks! I will do that :+1:

**Comment by dreadpool2** [comment]
@iamareebjamal Rebased my commits :+1:
